implementations/rust/daemon/src/vault.rs

```diff
@@ -0,0 +1,213 @@
+use std::path::PathBuf;
+
+use ockam_vault::{error::*, software::DefaultVault, types::*, Vault};
+
+use zeroize::Zeroize;
+
+pub struct FilesystemVault {
+    v: DefaultVault,
+    path: PathBuf,
+}
+
+impl FilesystemVault {
+    pub fn new(path: PathBuf) -> std::io::Result<Self> {
+        let create_path = path.clone();
+        std::fs::create_dir_all(create_path)?;
+
+        let mut vault = DefaultVault::default();
+        let to_secret = |path: PathBuf,
+                         data: &[u8]|
+         -> Result<(SecretKey, SecretKeyAttributes), VaultFailError> {
+            let attrs = SecretKeyAttributes {
+                xtype: SecretKeyType::Buffer(data.len()),
```

Comment:
Why would all the keys be `buffer`?

Reply:
@mikelodder7 - ah, they shouldn't be -- I meant to add a `TODO` here and ask how to check which key type the file would be... is there a byte or value in the stored key that indicates its type?

Reply:
Also, I don't believe they should all be for `purpose: SecretPurposeType::KeyAgreement`.. I might have to encode these bits of data into the file or the filename somehow.

Reply:
@mikelodder7  -- unless you suggest otherwise, I think I'm going to encode the following `usize` values as a part of the filename using these fn's below. For example, a key with context 1, of type `Aes256`, purpose `KeyAgreement` and persistence `Persistent` would be saved to disk named:

`./ockamd_vault/1_2_0_1` 


```rust
impl SecretKeyType {
    /// Convert enum to a number
    pub fn to_usize(&self) -> usize {
        match *self {
            SecretKeyType::Buffer(..) => 0,
            SecretKeyType::Aes128 => 1,
            SecretKeyType::Aes256 => 2,
            SecretKeyType::Curve25519 => 3,
            SecretKeyType::P256 => 4,
        }
    }

    /// Try to convert from a number to the rust enum
    pub fn from_usize(value: usize) -> Result<Self, VaultFailError> {
        match value {
            0 => Ok(SecretKeyType::Buffer(0)),
            1 => Ok(SecretKeyType::Aes128),
            2 => Ok(SecretKeyType::Aes256),
            3 => Ok(SecretKeyType::Curve25519),
            4 => Ok(SecretKeyType::P256),
            _ => Err(VaultFailErrorKind::InvalidParam(0).into()),
        }
    }
}
```

Reply:
or prefix/append them to the bytes in the file, but I think I'd prefer not to mess with the key itself. what do you think?

Reply:
In the OSX implementation I just take the `to_bytes` method for the attributes and put that as the first bytes with the key data after. We went over this once before but it was probably a lot of info at the time. 

See [here](https://github.com/ockam-network/ockam/blob/ml/channel-updates/implementations/rust/vault/src/osx/mod.rs#L197) and [here](https://github.com/ockam-network/ockam/blob/ml/channel-updates/implementations/rust/vault/src/osx/mod.rs#L149) for examples

Reply:
Thanks, @mikelodder7 - I do remember that now. I'll follow suit!